consensus/safety-rules/src/safety_rules.rs

```diff
@@ -445,19 +452,20 @@ impl TSafetyRules for SafetyRules {
 fn run_and_log<F, L, R>(callback: F, log_cb: L, log_entry: LogEntry) -> Result<R, Error>
 where
     F: FnOnce() -> Result<R, Error>,
-    L: Fn(StructuredLogEntry) -> StructuredLogEntry,
+    L: Fn(SafetyLogSchema) -> SafetyLogSchema,
 {
-    sl_debug!(log_cb(logging::safety_log(log_entry, LogEvent::Request)));
+    sl_debug!(log_cb(SafetyLogSchema::new(log_entry, LogEvent::Request)).into_struct_log());
     counters::increment_query(log_entry.as_str(), "request");
     callback()
         .map(|v| {
-            sl_info!(log_cb(logging::safety_log(log_entry, LogEvent::Success)));
+            sl_info!(log_cb(SafetyLogSchema::new(log_entry, LogEvent::Success)).into_struct_log());
             counters::increment_query(log_entry.as_str(), "success");
             v
         })
         .map_err(|err| {
-            sl_error!(log_cb(logging::safety_log(log_entry, LogEvent::Error))
-                .data(LogField::Message.as_str(), &err));
+            sl_error!(log_cb(SafetyLogSchema::new(log_entry, LogEvent::Error))
+                .error(err.clone())
```

Comment:
ahh, hahaha

Reply:
@gregnazario We don't have to clone things, this is a unique situation just due to the structure of this piece of code. As I mentioned above the proc-macro supports taking things by reference, I just punted on getting the lifetimes to work with these closures.